verification/curator-service/ui/src/components/ParsersAutocomplete.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+
+import axios from 'axios';
+import { Autocomplete } from '@material-ui/lab';
+import { CircularProgress, TextField } from '@material-ui/core';
+
+interface LambdaFunction {
+    name: string;
+}
+
+interface ParsersAutocompleteProps {
+    defaultValue: string;
+    onChange: (newValue: string) => void;
+}
+
+/** Autocomplete element that suggests parsers functions for automated ingestion setup */
+export default function ParsersAutocomplete(
+    props: ParsersAutocompleteProps,
+): JSX.Element {
+    const [open, setOpen] = React.useState(false);
+    const [options, setOptions] = React.useState<string[]>([]);
+    const loading = open && options.length === 0;
+
+    React.useEffect(() => {
+        let active = true;
+
+        if (!loading) {
+            return undefined;
+        }
+
+        (async () => {
+            const res = await axios.get<LambdaFunction[]>(
+                '/api/sources/parsers',
+            );
+
+            if (active) {
+                setOptions(res.data.map((f) => f.name));
+            }
+        })();
+
+        return () => {
+            active = false;
+        };
+    }, [loading]);
+
+    React.useEffect(() => {
+        if (!open) {
+            setOptions([]);
+        }
+    }, [open]);
+
+    return (
+        <Autocomplete
+            style={{ width: 300 }}
+            open={open}
+            // freeSolo useful in tests or if a parsing function wasn't built yet.
```

Comment:
Not sure it's worth adding if it's just for tests. Is testing a PITA without this? I don't think we'll want to allow preemptive setting of the parser, either.

Reply:
testing is unfeasible without this as we don't currently mock the calls to the lambda service they just fail and the options are empty.
Maybe it's time we think about how to properly do that in integration tests as we did for geocoding (providing a fake might make sense)